File: 백준/Bronze/5073. 삼각형과 세 변/삼각형과 세 변.py
while True:
    t=[]
    t=list(map(int,input().split()))
    if t[0]==0 and t[1]==0 and t[2]==0:
        break
        
    t.sort()
    if t[0]+t[1]<=t[2]:
        print("Invalid")
    elif t[0]==t[1] and t[1]==t[2]:
        print("Equilateral")
    elif t[0]==t[1] or t[1]==t[2] or t[2]==t[0]:
        print("Isosceles")
    else:
        print("Scalene")

File: 백준/Bronze/5073. 삼각형과 세 변/삼각형과 세 변.java
import java.util.*;
import java.io.*;

class Main{
    public static void main(String[] args) throws IOException{
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringBuilder sb = new StringBuilder();
        
        while(true){
            StringTokenizer st = new StringTokenizer(br.readLine());
            int a = Integer.parseInt(st.nextToken());
            int b = Integer.parseInt(st.nextToken());
            int c = Integer.parseInt(st.nextToken());
            if(a==0 && b==0 && c==0) break;
            int max = Math.max(a, Math.max(b,c));
                
            if(a==b && b==c) sb.append("Equilateral");
            else if(max>=(a+b+c-max)) sb.append("Invalid");
            else if (a!=b && b!=c && a!=c) sb.append("Scalene");
            else sb.append("Isosceles");
            sb.append("\n");
        }
        
        System.out.println(sb);
		    
    }
}